src/codecs/tiff.rs

```diff
@@ -28,7 +28,7 @@ where
 {
     dimensions: (u32, u32),
     color_type: ColorType,
-    inner: tiff::decoder::Decoder<R>,
+    inner: Option<tiff::decoder::Decoder<R>>,
```

Comment:
Please add a comment that this is purely so the value can be replaced behind a mutable `&mut self` via `Option::take`.